app/src/common/shared/com/igalia/wolvic/utils/BitmapCache.java

```diff
@@ -32,6 +33,16 @@ public class BitmapCache {
     private Surface mCaptureSurface;
     private boolean mCapturedAcquired;
 
+    private String normalizeKey(@NonNull String input) {
+        // DiskLruCache keys must match the regex [a-z0-9_-]{1,64}
+        final int MAX_LENGTH = 64;
+        CRC32 crc = new CRC32();
+        crc.update(input.getBytes());
+
+        String key = Long.toHexString(crc.getValue()) + '_' + input.toLowerCase().replaceAll("[^a-z0-9_-]", "_");
+        return key.substring(0, Math.min(key.length(), MAX_LENGTH));
```

Comment:
Why not always using MessageDigest.getIntace("SHA-256") for generating the hash? That always generates a 64-hex digit hash which seems is what you need